src/composeappmanager.cc

```diff
@@ -310,3 +310,17 @@ void ComposeAppManager::handleRemovedApps(const Uptane::Target& target) const {
 }
 
 std::string ComposeAppManager::getCurrentHash() const { return sysroot_->getCurDeploymentHash(); }
+
+std::string ComposeAppManager::containerDetails() const {
+  std::string cmd = cfg_.docker_bin.string();
+  cmd +=
+      " ps --format 'App({{.Label \"com.docker.compose.project\"}}) Service({{.Label "
+      "\"com.docker.compose.service\"}} {{.Label \"io.compose-spec.config-hash\"}})'";
```

Comment:
I still don't get what the value brings "io.compose-spec.config-hash". IMHO reporting of the installed apps' uri that includes sha/hash/digest would suffice.
My point is that aklite needs to report what it actually has installed, e.g. Target name, rootfs ostree hash and uri of apps it downloaded and installed. In this, case a user will see what was actually pulled and installed and can compare uris. In the case of the recent issue, it would report zero apps' uri since zero of them was downloaded and installed.

Checking out what is actually running is also useful but I think it's too much for the update agent, I think this is the responsibility of some other tool responsible for monitoring, a set of running apps/containers can change under some other than an update circumstances. 

In any case, I don't get why container images are labeled with "config-hash" not by app uri?

Reply:
The api-uri's hash is dependent on the content of docker-compose.yml. So you need something else. You could just put "build-number" or something. I just thought putting a value that's actually verifiable from both directions (what the target says and what the engine shows) could be helpful.

Reply:
thanks for the explanation, it all makes sense to me now. Let's merge it as it now or replace config-hash with a build/Target number as we discussed today and then merge. Thanks.

Reply:
let me add logic to emit what we think we installed.

Reply:
it now looks like
~~~
$ fioctl devices updates ostree-qemu 232-0df4f8ba-0de6-4a83-8074-b2dc08b760e3
2021-01-28T18:07:20+00:00 : EcuDownloadStarted(intel-corei7-64-lmp-232)
 Details:
 | Updating from intel-corei7-64-lmp-231 to intel-corei7-64-lmp-232
2021-01-28T18:07:22+00:00 : EcuDownloadCompleted(intel-corei7-64-lmp-232) -> Succeed
2021-01-28T18:07:22+00:00 : EcuInstallationStarted(intel-corei7-64-lmp-232)
2021-01-28T18:07:34+00:00 : EcuInstallationCompleted(intel-corei7-64-lmp-232) -> Succeed
 Details:
 | OSTree hash already installed, same as current
 | # Apps installed:
 | hub.foundries.io/andy-corp/homelab@sha256:59e239baa4b1f30ba01ed0425b8819dc1214e788ae83d6d027ad4cea326a193a
 | hub.foundries.io/andy-corp/shellhttpd@sha256:c334fc25c77699f2a849b674646b45181f1faf57a77029bfbcee81902368cd59
 | # Apps running
 | App(shellhttpd) Service(httpd 9a8e3b29be3526ab0516819a37ea1339816eb59b55bbe6b86f0cfbe57c9679a4)
 | App(homelab) Service(sshkey-sync 84f09b06bee04d8c4df7e7363482d3c7c4de65c95e6d0bdf0da1d0699380b049)
~~~